### PR TITLE
Minor update to addEntrypoint.bash

### PR DESCRIPTION
**Pull request by WarpWing**
**Pull Request Description**

Starts to integrate feedback from #25. Still working on some minor adjustments.
_Include a "Closes #" line for the issue your PR closes._

---

**Licensing Certification**

FarmData2 is a [Free Cultural Work](https://freedomdefined.org/Definition) and all accepted contributions are licensed as described in the LICENSE.md file. This requires that the contributor holds the rights to do so. By submitting this pull request **I certify that I satisfy the terms of the [Developer Certificate of Origin](https://developercertificate.org/)** for its contents.

### CODE REVIEW

**Comment by WarpWing** [comment]
I've added a check for a clean tree and now the script no longer switches to the development branch. It fetches the latest updates for the development branch from the origin and updates the local development branch without checking it out.

File: bin/addEntrypoint.bash [hunk 1]
```diff
@@ -6,28 +6,37 @@ source lib.bash
...
-if [[ ! "$GIT_STATUS" =~ ^"nothing to commit, working tree clean"$ ]]; then
+GIT_STATUS=$(git status --porcelain)
+if [ -n "$GIT_STATUS" ]; then
   echo -e "${ON_RED}ERROR:${NO_COLOR} The working tree must be clean to add an entry point."
   echo "Commit changes to a feature branch or use git stash."
   echo "Then run this script again."
   exit 255
 fi
 
+# Get the current Git branch
```
Comment (braughtg) [inline bin/addEntrypoint.bash:22]: Seems like if development is not active then it will be updated twice now.

File: bin/addEntrypoint.bash [hunk 2]
```diff
@@ -51,7 +60,7 @@ echo ""
 
 DRUPAL_ROUTE="$DRUPAL_ROUTE_PREFIX""\/$ENTRY_POINT"
 ENTRY_POINT_SRC_DIR="$MODULE_DIR/src/entrypoints/$ENTRY_POINT"
-ENTRY_POINT_TEMPLATE_DIR="$SCRIPT_DIR/templates/entrypoints"
+ENTRY_POINT_TEMPLATE_DIR="$SCRIPT_DIR/templates/entrypoint" # Updated directory name
```
Comment (braughtg) [inline bin/addEntrypoint.bash:63]: Be sure the PR also renames the directory itself.

File: bin/addEntrypoint.bash [hunk 3]
```diff
@@ -65,17 +74,13 @@ if [[ ! "$FEATURE_BRANCH_EXISTS" == "" ]]; then
...
-# Check if the directory for the entry point exits...
+# Check if the directory for the entry point exists...
 if [ -d "src/entrypoints/$ENTRY_POINT" ]; then
   echo -e "${ON_RED}ERROR:${NO_COLOR} A directory for the entry point $ENTRY_POINT already exists"
-  echo "in the directory $$ENTRY_POINT_SRC_DIR."
+  echo "in the directory $ENTRY_POINT_SRC_DIR."
   echo "Pick a different name for your entry point."
   echo "OR:"
   echo "  Remove the src/entrypoints/$ENTRY_POINT directory"
-  echo "  And remove any definitions related to the entry point $ENTRY_POINT from the files:"
```
Comment (braughtg) [inline bin/addEntrypoint.bash:75]: I think these lines need to remain as a reminder that simply removing the directory is not sufficient.

File: bin/addEntrypoint.bash [hunk 4]
```diff
@@ -65,17 +74,13 @@ if [[ ! "$FEATURE_BRANCH_EXISTS" == "" ]]; then
...
 fi
 
-# Check if the directory for the entry point exits...
+# Check if the directory for the entry point exists...
 if [ -d "src/entrypoints/$ENTRY_POINT" ]; then
   echo -e "${ON_RED}ERROR:${NO_COLOR} A directory for the entry point $ENTRY_POINT already exists"
-  echo "in the directory $$ENTRY_POINT_SRC_DIR."
+  echo "in the directory $ENTRY_POINT_SRC_DIR."
   echo "Pick a different name for your entry point."
   echo "OR:"
```
Comment (braughtg) [inline bin/addEntrypoint.bash:82]: Probably clarify here:  "OR if you are sure you want to replace the entry point:"

File: bin/addEntrypoint.bash [hunk 5]
```diff
@@ -125,6 +130,11 @@ select ENTRY_POINT_PARENT in "${MENUS[@]}"; do
 done
 echo ""
 
+# Create the feature branch
```
Comment (braughtg) [inline bin/addEntrypoint.bash:133]: I think the feature branch should still be created after the confirmation below.

**Review by braughtg — COMMENTED** [review]
This looks like a good start.  I know some of what you have is wip, but added comments anyway just to be sure.  Disregard those that don't apply in areas where your code was/is still evolving.

**Comment by braughtg** [comment]
I am currently working on other changes to the `addEntrypoint` script.  My current plan is to reduce merge conflicts by incorporating your changes from this PR into my PR and adding you as a co-author for that commit.

**Pull request closed without merging**